lib/resty/websocket/client.lua

```diff
@@ -98,9 +98,17 @@ function _M.connect(self, uri, opts)
         path = "/"
     end
 
-    local ssl_verify, proto_header, origin_header, sock_opts = false
+    local ssl_verify, proto_header, origin_header, custom_header, sock_opts = false
 
     if opts then
+        local headers = opts.headers
+        if headers then
+            custom_header = ""
+            for field,value in pairs(headers) do
```

Comment:
Maybe it's better to use a flat array for this option, as in

`{ name1, value1, name2, value2, ...}`.

So that we can avoid iterating through a hash table. Additionally, iterating through a hash table is not JIT compiled in LuaJIT.